crates/tabby-scheduler/tests/integration_test.rs

```diff
@@ -0,0 +1,30 @@
+use tabby_scheduler::{dataset, index, repository};
+
+#[cfg(test)]
+mod tests {
+    use tabby_common::{
+        config::{Config, Experimental, Repository},
+        path::set_tabby_root,
+    };
+    use temp_testdir::*;
+    use tracing_test::traced_test;
+
+    use super::*;
+
+    #[traced_test]
+    #[test]
+    fn end_to_end() {
+        set_tabby_root(TempDir::default().to_path_buf());
+
+        let config = Config {
+            repositories: vec![Repository {
+                git_url: "https://github.com/TabbyML/interview-questions".to_owned(),
+            }],
+            experimental: Experimental::default(),
+        };
+
+        repository::sync_repositories(&config).unwrap();
```

Comment:
might simply call `tabby_scheduler::scheduler(now = true)`?

by doing so we can keep dataset / index / repository package private

Reply:
Sure but how do we pass in the config then? Writing to a local config.toml file?

Reply:
Or we could add a `config: Option<Config>` param to `scheduler()` function. Then in `main()` we pass `None` to ask `scheduler()` to load actual config file in real use case.

Reply:
Adding param rather than writing local file, in order to avoid actual fs io in integ test.